src/main/resources/templates/oidcRedirect.html

```diff
@@ -14,120 +14,257 @@
+            </div>
+            <div class="form-group">
+                <label for="prompt">prompt:</label>
+                <input type="text" id="prompt" name="prompt" th:value="${prompt}" class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="redirect_uri">redirect_uri:</label>
+                <input type="text" id="redirect_uri" name="redirect_uri" th:value="${redirect_uri}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="response_mode">response_mode:</label>
+                <input type="text" id="response_mode" name="response_mode" th:value="${response_mode}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="response_type">response_type:</label>
+                <input type="text" id="response_type" name="response_type" th:value="${response_type}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="scope">scope:</label>
+                <input type="text" id="scope" name="scope" th:value="${scope}" class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="state">state:</label>
+                <input type="text" id="state" class="form-control" name="state" th:value="${state}" />
+            </div>
+            <input id="post_button" type="submit" value="Submit POST" class="btn btn-primary" disabled="disabled">
+        </form>
+    </div>
 
-<br/>
+    <br />
 
-<button class="accordion large-font">
-    Click to view the GET version of Step 2 of the LTI Core Standard Launch
-</button>
-<div class="panel">
-    <div class="alert alert-primary">GET URL: <span th:text="${oidcEndpointComplete}"/></div>
-</div>
-<a th:href="${oidcEndpointComplete}" class="btn btn-primary" role="button" aria-pressed="true">Perform GET as Step 2</a>
+    <button class="accordion large-font">
+        Click to view the GET version of Step 2 of the LTI Core Standard Launch
+    </button>
+    <div class="panel">
+        <div class="alert alert-primary">GET URL: <span th:text="${oidcEndpointComplete}" /></div>
+    </div>
+    <a th:href="${oidcEndpointComplete}"  id="get_button" class="btn btn-primary disabled" role="button" aria-pressed="true">Perform GET as Step
+        2</a>
+
+<script th:inline="javascript">
+
+    const nonceStateCheck = async () => {
+
+    // placeholder for state and nonce validations
+    let stateValid = false, nonceValid = false, stateNonceValid = false;
+
+    // state and nonce values. These should be generated on the server side
+    const stateHash = [[${ state }]];
+    const nonce = [[${ nonce }]];
+
+    // origin of the LTI launch. Currently, * is the only value the LMS appears to accept
+    let origin = [[${oidc_authorization_uri }]];
+    //let origin = '*';
```

Comment:
Should this be deleted? Wasn't this line actually needed for Canvas? Is this code block getting used still?

Reply:
It is not needed for canvas, I found the right way to call canvas with the oidc_authorization_uri (It needs to remove the values after the / so I do that in the controller before passing the value here)
We can remove the "*" line